qualtran/bloqs/mod_arithmetic/mod_division_test.py

```diff
@@ -0,0 +1,86 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import math
+
+import pytest
+import sympy
+
+import qualtran.testing as qlt_testing
+from qualtran.bloqs.mod_arithmetic.mod_division import _kaliskimodinverse_example, KaliskiModInverse
+from qualtran.resource_counting import get_cost_value, QECGatesCost
+from qualtran.resource_counting.generalizers import ignore_alloc_free, ignore_split_join
+
+
+@pytest.mark.parametrize('bitsize', [5, 6])
+@pytest.mark.parametrize('mod', [3, 5, 7, 11, 13, 15])
+def test_kaliski_mod_inverse_classical_action(bitsize, mod):
+    blq = KaliskiModInverse(bitsize, mod)
+    cblq = blq.decompose_bloq()
+    p2 = pow(2, bitsize, mod)
+    for x in range(1, mod):
+        if math.gcd(x, mod) != 1:
+            continue
+        x_montgomery = (x * p2) % mod
+        inv_x = pow(x, -1, mod)
+        inv_x_montgomery = (inv_x * p2) % mod
```

Comment:
can you use @fpapa250 's new utilities `QMontgomery.to_montgomery()` or whatever it's called to make this more clear/explicit?

Reply:
done